### DISCUSSION TITLE
[Binary Caching] Add cleanup feature of old/deprecated archives

### DISCUSSION THREAD

**Discussion by julianxhokaxhiu** [microsoft/vcpkg#19589] [Ideas]
****

**Is your feature request related to a problem? Please describe.**
vcpkg is awesome, it helps you to finally define your dependencies through a manifest, and manage them very easily through CMake. But there's a problem: when moving forward with your project, by simply updating your tooling or moving forward with your own manifest file, new cached archives will be produced and old ones will settle there, forever, unless manually deleted.

It would be nice if vcpkg had a way to cleanup those automatically, through a dedicated action in the command line.

**Proposed solution**
Upon building new cached archives for a given port, old ones are marked as "deprecated". vcpkg keeps track of this info somewhere and then through a command line tool you could cleanup those, using something like `vcpkg cache clean`.

An additional parameter `--force` might be introduced, which would cleanup ALL the cache, including cached archives which are not marked as deprecated.

**Describe alternatives you've considered**
At the moment I'm deleting the folder in `%localappdata%\vcpkg` from time to time and rebuilding the entire ports from scratch, but it's far from optimal especially if you have ports which takes quite some time to be built ( ffmpeg for eg. )

**Additional context**
N/A

**Comment by Be-ing** [comment]
Related: setting a size limit for the cache https://github.com/microsoft/vcpkg/discussions/19452

**Comment by julianxhokaxhiu** [comment]
Hello vcpkg team, has there been some progress towards this direction? Any help needed in case?

Thank you in advance,
Julian

**Reply by Osyotr** [reply]
https://github.com/microsoft/vcpkg-tool/pull/325

**Comment by martingalvan-volue** [comment]
Hi, are there any updates regarding this? The PR linked by @Osyotr has been in draft state for 3 years now, and was later superseded by https://github.com/microsoft/vcpkg-tool/pull/945 which has been stale for 2 years (and IMHO is not an optimal solution to this problem).

**Reply by autoantwort** [reply]
Yeah microsoft does not want to spend time/money on reviewing PRs so they got stale.
